RocketChat/Rocket.Chat
# fix: resolve issue with emoji identifiers in headline markdown not being evaluated

## Proposed changes (including videos or screenshots)
This fix addresses an issue where emoji identifiers (e.g., `:smile:`) in markdown headlines were not being correctly evaluated and rendered as emojis. Previously, when an emoji identifier was used in a heading (e.g., `# :smile: Heading`), it remained as plain text rather than being converted into the corresponding emoji.

### Changes:
- Updated the markdown parsing logic to ensure emoji identifiers are evaluated and converted into emojis when used in headline markdown (`#`, `##`, etc.).
- Added tests to verify that emojis in headlines are now properly rendered.
- Improved handling of edge cases, such as when emojis are used with other formatting elements in headings.

### Impact:
- Headline markdown that includes emoji identifiers will now correctly display emojis, improving readability and user experience.

---

## Issue(s)
Resolves the issue of emoji identifiers not being processed in markdown headlines.

---

## Steps to test or reproduce
1. Create a markdown heading like `# :smile: Heading`.
2. Ensure that the emoji is correctly rendered within the headline.

---

## Further comments
This update improves the user experience by making markdown headings more visually engaging through the proper handling of emoji identifiers.
